processing/src/main/java/io/druid/segment/incremental/IncrementalIndex.java

```diff
@@ -556,9 +541,9 @@ public DimensionHolder getDimValues()
     return dimValues;
   }
 
-  public List<String> getDimensions()
+  public String[] getDimensions()
   {
-    return dimensions;
+    return dimensions.toArray(new String[dimensions.size()]);
```

Comment:
Yes, it's all for handling generics in IncrementalIndex, which seemed not needed anymore. I'll remove generics on it.